torchrl/modules/tensordict_module/sequence.py

```diff
@@ -118,27 +118,18 @@ def __init__(
         in_keys = []
         out_keys = []
         for module in modules:
-            in_keys += module.in_keys
+            # we sometimes use in_keys to select keys of a tensordict that are
+            # necessary to run a TensorDictModule. If a key is an intermediary in
+            # the chain, there is no reason why it should belong to the input
+            # TensorDict.
+            in_keys += [key for key in module.in_keys if key not in out_keys]
```

Comment:
these changes should not be part of this PR

Reply:
Once you update your branch from main after #261 is merged this should disappear